src/TypeSpec.Extension/Emitter.Csharp/package.json

```
@@ -50,18 +62,23 @@
     },
     "dependencies": {
         "@autorest/csharp": "3.0.0-beta.20230309.1",
-        "@azure-tools/typespec-azure-core": "0.35.0",
-        "@azure-tools/typespec-client-generator-core": "0.35.0",
-        "@typespec/compiler": "0.49.0",
-        "@typespec/eslint-config-typespec": "0.49.0",
-        "@typespec/eslint-plugin": "0.49.0",
-        "@typespec/http": "0.49.0",
-        "@typespec/library-linter": "0.49.0",
-        "@typespec/rest": "0.49.0",
-        "@typespec/versioning": "0.49.0",
         "fs-extra": "^10.1.0",
         "js-yaml": "^4.1.0",
         "json-serialize-refs": "0.1.0-0",
         "winston": "^3.8.2"
+    },
+    "peerDependencies": {
+        "@azure-tools/typespec-azure-resource-manager": ">=0.35.0 <1.0.0",
```

```suggestion
        "@azure-tools/typespec-azure-resource-manager": ">=0.36.0 <1.0.0",
```

I think you'd want to set the min version to the current latest, unless you intentionally want to support version N-1 of these deps.

---

I think n-1 is best since we don't know if all emitters have updated and we want to be compatible

---

But this means you also can only use features in (n-1), and to verify this, it would be best to set your devDep to (n-1).

---

I pushed an update to use latest as min peerDep, which matches autorest-python.  I'm pretty sure this is what you want, so you can use the latest features in your dependencies.